### PR TITLE
Frontend changes for theory_str

### PR DESCRIPTION
**Pull request by mtrberzi**
This patch makes a small number of frontend changes (`seq_decl_plugin`, `seq_rewriter`) in preparation for the `theory_str` merge.

### CODE REVIEW

File: src/ast/seq_decl_plugin.cpp [hunk 1]
```diff
@@ -573,7 +573,7 @@ void seq_decl_plugin::set_manager(ast_manager* m, family_id id) {
     m_char = bv.mk_sort(8);
     m->inc_ref(m_char);
     parameter param(m_char);
-    m_string = m->mk_sort(symbol("String"), sort_info(m_family_id, SEQ_SORT, 1, &param));
+    m_string = m->mk_sort(symbol("StringSequence"), sort_info(m_family_id, SEQ_SORT, 1, &param));
```
Comment (NikolajBjorner) [inline src/ast/seq_decl_plugin.cpp]: Seems it should be "String". We will end up printing these types. _STRING_SORT gets used by the parser front-end.

File: src/ast/seq_decl_plugin.h [hunk 2]
```diff
@@ -273,6 +273,15 @@ class seq_util {
         bool is_in_re(expr const* n)    const { return is_app_of(n, m_fid, OP_SEQ_IN_RE); }
         bool is_unit(expr const* n)     const { return is_app_of(n, m_fid, OP_SEQ_UNIT); }
 
+        bool is_string_term(expr const * n) const {
+            sort * s = get_sort(n);
+            return (u.is_seq(s) && u.is_string(s));
```
Comment (NikolajBjorner) [inline src/ast/seq_decl_plugin.h]: u.is_seq should be redundant.

**Comment by NikolajBjorner** [comment]
looks mostly good, main question is about "StringSequence"

**Comment by mtrberzi** [comment]
I had some reason for needing to change that name. Let me look into that tomorrow.

File: src/ast/seq_decl_plugin.h (hunk 2 above)
Comment (mtrberzi) [inline src/ast/seq_decl_plugin.h]: Right, `is_string` does make that check.

**Comment by mtrberzi** [comment]
Actually it seems to work with "String" instead of "StringSequence", I really can't remember why I thought I needed to do that but if it comes up again I'll try to work around it.

**Pull request merged**